backends/simplego/gen_exec_binary.go

```diff
@@ -40,7 +40,6 @@ func execAdd(backend *Backend, node *Node, inputs []*Buffer, inputsOwned []bool)
 	lhs, rhs, output, lhsIsScalarOr1, rhsIsScalarOr1 := binaryOperandsAndOutput(backend, inputs, inputsOwned, node.shape) // Add is commutative, so if any of the two is scalar, make the rhs the scalar one.
 	if lhsIsScalarOr1 && !rhsIsScalarOr1 {
 		lhs, rhs = rhs, lhs
-		lhsIsScalarOr1, rhsIsScalarOr1 = rhsIsScalarOr1, lhsIsScalarOr1
```

Comment:
It's an interesting case this one:

By one hand it's true that `lhsIsScalarOr1`, `rhsIsScalarOr1` are no longer used, so this assignment is a no-op.

But ... unfortunately, `lhsIsScalarOr1` and `rhsIsScalarOr1` variables keep existing, and in the future one may use them, unaware that their value is now (after removing this line) __occasionally__ incorrect.

I'm a very "defensive" coder when I can. My intuition is that I would leave this assignment here just in case (it's probably optimized away by the compiler anyway). Or find a way to make these variables no longer live after this if-clause.

Any thoughts ?